File: 2017 czerwiec/c++/1.cpp
#include <iostream>
#include <fstream>
using namespace std;

bool is_prime[10001];

int main()
{
    ifstream in("punkty.txt");

    for (int i = 2; i <= 10000; i++)
        is_prime[i] = true;
    
    for (int i = 2; i <= 10000; i++)
    {
        if (is_prime[i])
        {
            for (int j = i * i; j <= 10000; j += i)
            {
                is_prime[j] = false;
            }
        }
    }

    int ans = 0;
    for (int i = 1; i <= 1000; i++)
    {
        int x, y;
        in >> x >> y;
        ans += (is_prime[x] && is_prime[y]);
    }
    cout << ans << '\n';
}

File: 2017 czerwiec/python/1.py
with open("punkty.txt", "r") as f:
    p = f.read().strip().split('\n')

def is_prime(n):
    if n <= 1:
        return False
    for i in range(2, n):
        if n % i == 0:
            return False
    return True

ans = 0
for pair in p:
    x, y = list(map(int, pair.split()))
    if is_prime(x) and is_prime(y):
        ans += 1
print(ans)